### DISCUSSION TITLE
Do I need a streaming service for quality video playback or can I simply reference videos in google cloud storage for quality playback?

### DISCUSSION THREAD

**Discussion by Zelfapp** [jhomlala/betterplayer#723] [General]
New to video in general. BetterPlayer is fantastic. Being new to video I'm at a fork for deciding the proper way to playback pre-recorded videos that are a maximum of 2 minutes in length?

My flutter app currently downloads pre-recorded videos and then plays them once they've downloaded in BetterPlayer.

I want the user experience to be similar to scrolling through videos on the YouTube app and you click on them and they playback fast at good quality.

Do I need to look into a service like agoria.io to stream my pre-recorded videos where they are encoded an optimized?

Or will I get as good an experience simply setting the `network` data source in betterplayer to google cloud storage where my videos (mp4's) are stored? The videos do not have any special encoding or optimization.

What are pros out there doing where they need to keep the costs as low as possible for video storage and for displaying and viewing videos in a YouTube app similar fashion?

**Comment by jhomlala** [comment]
You can check services like Google Cloud, Amazon S3 etc. but you need to remember that if you want to have low latency and generally fast as possible playback then you will pay a lot for these services. There's no golden rule here. You should experiment with different providers to see what's the best hosting service. Definitely the best way is to have some video hosting and service videos with network, rather than downloading it on user's device.
